react/features/breakout-rooms/actions.js

```diff
@@ -0,0 +1,184 @@
+            breakoutRoomJid
+        };
+
+        getCurrentConference(getState).sendMessage(message, 'focus');
+    };
+}
+
+
+/**
+ * Action to auto-assign the participants to breakout rooms.
+ *
+ * @returns {Function}
+ */
+export function autoAssignToBreakoutRooms() {
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        const { rooms = {} } = getBreakoutRooms(getState);
+        const breakoutRooms = _.filter(rooms, (room: Object) => !room.isMainRoom);
+
+        if (breakoutRooms) {
+            const participantIds = Array.from(getRemoteParticipants(getState).keys());
+            const length = Math.ceil(participantIds.length / breakoutRooms.length);
+
+            _.chunk(_.shuffle(participantIds), length).forEach((group, index) =>
+                group.forEach(participantId =>
+                    dispatch(sendParticipantToRoom(participantId, breakoutRooms[index].id))
+                )
+            );
+        }
+    };
+}
+
+/**
+ * Action to send a participant to a room.
+ *
+ * @param {string} participantId - The participant id.
+ * @param {string} roomId - The room id.
+ * @returns {Function}
+ */
+export function sendParticipantToRoom(participantId: string, roomId: string) {
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        const conferenceOptions = getConferenceOptions(getState);
+        const id = participantId.indexOf('@') >= 0
+            ? participantId
+            : `${getCurrentRoomId(getState)}@${isInBreakoutRoom(getState)
+                ? `breakout.${conferenceOptions.hosts.domain}`
+                : conferenceOptions.hosts.muc
+            }/${participantId}`;
+        const message = {
+            type: JSON_TYPE_MOVE_TO_ROOM_REQUEST,
+            roomId
+        };
+
+        getCurrentConference(getState).sendMessage(message, id);
+    };
+}
+
+/**
+ * Action to move to a room.
+ *
+ * @param {string} roomId - The room id to move to. If omitted move to the main room.
+ * @returns {Function}
+ */
+export function moveToRoom(roomId?: string) {
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        // FIXME: Move this code to conference.js
+        // respectively the base/conference feature.
+        const _roomId = roomId || getMainRoomId(getState);
+
+        if (navigator.product === 'ReactNative') {
+            const conference = getCurrentConference(getState);
+            const { audio, video } = getState()['features/base/media'];
+
+            dispatch(conferenceWillLeave(conference));
+            conference.leave()
+            .catch(error => {
+                logger.warn(
+                    'JitsiConference.leave() rejected with:',
+                    error);
+
+                dispatch(conferenceLeft(conference));
+            });
+            dispatch(clearNotifications());
+            dispatch(setRoom(_roomId));
+            dispatch(createConference()).then(result => {
+                dispatch(setAudioMuted(audio.muted));
```

Comment:
Why does this need to be specified?

Reply:
Good catch. This is wrong. Actually createConference does not return a promise. Additionally there is no need to reset the muted state. Removed this.

Reply:
Understand. appNavigate is too heavy: it disconnects and reconnects while we maintain the connection (this way the client keeps its jid and affiliation in the conference).